web/src/stream_popover.js

```diff
@@ -562,6 +564,64 @@ export async function build_move_topic_to_stream_popover(
         $("#move_topic_form .move_messages_edit_topic").trigger("focus");
     }
 
+    // The following logic is correct only when
+    // both message_lists.current.data.fetch_status.has_found_newest
+    // and message_lists.current.data.fetch_status.has_found_oldest are true;
+    // otherwise, we cannot be certain of the correct count.
+    function get_count_of_messages_to_be_moved(selected_option) {
+        if (selected_option === "change_one") {
+            return 1;
+        }
+        if (selected_option === "change_later") {
+            return message_util.get_count_of_messages_in_topic_sent_after_current_message(
+                current_stream_id,
+                topic_name,
+                message.id,
+            );
+        }
+        return message_util.get_messages_in_topic(current_stream_id, topic_name).length;
+    }
+
+    function update_move_messages_count_text(selected_option) {
+        const terms = narrow_state.filter().terms();
+        const message_move_count = get_count_of_messages_to_be_moved(selected_option);
+        const has_found_newest = message_lists.current.data.fetch_status.has_found_newest();
+        const has_found_oldest = message_lists.current.data.fetch_status.has_found_oldest();
+        const narrowed_stream_id = terms[0].operand;
+        const narrowed_topic_name = terms[1].operand;
```

Comment:
There are already helper functions in `narrow_state` to get the current stream ID and topic name.